p2p/conn/connection.go

```diff
@@ -121,8 +121,8 @@ func (cfg *MConnConfig) maxPacketMsgTotalSize() int {
 }
 
 // DefaultMConnConfig returns the default config.
-func DefaultMConnConfig() *MConnConfig {
-	return &MConnConfig{
+func DefaultMConnConfig() MConnConfig {
```

Comment:
all our `Default..Config` functions return pointers. Maybe we should be consistent

Reply:
Using pointers turned out to be racy, in general I would challenge that those should be mutable. Having them non-pointers would communicate that better. On other other hand we can always dereference them where they are consumed.